eclass/dist-kernel-utils.eclass

```diff
@@ -104,12 +104,23 @@ dist-kernel_install_kernel() {
 	if [[ ${magic} == MZ ]]; then
 		einfo "Combined UEFI kernel+initramfs executable found"
 		# install the combined executable in place of kernel
-		image=${initrd}.uefi
+		image=${initrd}.efi
 		mv "${initrd}" "${image}" || die
-		# put an empty file in place of initrd.  installing a duplicate
-		# file would waste disk space, and removing it entirely provokes
-		# kernel-install to regenerate it via dracut.
-		> "${initrd}"
+		# We moved the generated initrd, prevent dracut from running again
+		local plugins=()
+		for file in "${EROOT}"/usr/lib/kernel/install.d/*; do
+			if [[ ${file} != */50-dracut.install && \
+				${file} != */51-dracut-rescue.install && \
+				${file} == *.install ]]; then
+					plugins+=( "${file}" )
+			fi
+		done
+		for file in "${EROOT}"/etc/kernel/install.d/*; do
```

Comment:
Ditto.

Reply:
We also need to `shopt -s nullglob` here, otherwise this breaks completely if there are no matches.